packages/app-project/src/screens/ProjectHomePage/getServerSideProps.js

```diff
@@ -0,0 +1,18 @@
+import { projects } from '@zooniverse/panoptes-js'
+import fetchWorkflowsHelper from '@helpers/fetchWorkflowsHelper'
+
+export async function getServerSideProps({ params, query, req, res }) {
+  const { owner, project } = params
+  const { env } = query
+  const projectQuery = {
+    env,
+    slug: `${owner}/${project}`
+  }
+  const response = await projects.get({ query: projectQuery })
+  const [projectData] = response.body.projects
+  const language = projectData.primary_language
+  const { active_workflows, default_workflow } = projectData.links
+  const workflows = await fetchWorkflowsHelper(language, active_workflows, default_workflow, env)
+  const props = { workflows }
+  return ({ props })
+}
```

Comment:
What if these requests fail? What kind of handling should these have for different scenarios, say 404 vs 500s?

Reply:
Good question. What happens at the moment? Doesn't the component display a message about workflows failing to load?

Reply:
There's a story for it: https://zooniverse.github.io/front-end-monorepo/@zooniverse/fe-project/index.html?path=/story/project-app-screens-project-home-hero--error

Reply:
We could send a project workflows error state back with the page props. The generated HTML might be cached across page refreshes, which could be a problem if it's a temporary error. 

Uncaught errors on the server will generate a 500 response, but that won't be cached. The error page is customisable, so setting up a project error page might be a good first issue. That ties into having a custom 404 page too.

Reply:
NextJS recommend not handling server-side errors, although there's next-to-no official documentation about that, just discussions about best practice. https://github.com/vercel/next.js/issues/11644 has some discussion about error-handling.

If we do handle errors, and send back a 200 response with a project home page, then it's worth noting that this can be indexed by search engines (as well as cached by browsers.)

Reply:
Next 10 adds [`notFound` and `redirect` keys](https://nextjs.org/blog/next-10#redirect-and-notfound-support-for-getstaticprops--getserversideprops) to the returned object from `getServerSideProps`. That could be useful.

Also worth noting that the workflows request can't 404. `notFound` would only be needed when requesting a specific resource.

Reply:
I'm fine with punting this issue for version 10 if it has a better API for it. Let's track in an issue?